src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1924,14 +1930,22 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
 
                 statements = ImmutableArray<BoundStatement>.Empty;
             }
-            else if (body == null)
-            {
-                statements = ImmutableArray.Create(constructorInitializer);
-            }
             else
             {
-                statements = ImmutableArray.Create(constructorInitializer, body);
-                originalBodyNested = true;
+                if (constructorInitializer.WasCompilerGenerated && constructorInitializer is BoundExpressionStatement { Expression: { } constructorExpression })
+                {
+                    ReportCtorInitializerCycles(method, constructorExpression, compilationState, diagnostics);
```

Comment:
>ReportCtorInitializerCycles(method, constructorExpression, compilationState, diagnostics); [](http://example.com/codeflow?start=20&length=90)

I am not sure why this check was taken out of ```BindImplicitConstructorInitializerIfAny```. Are there cases where the ```constructorInitializer``` here is not produced by it?